logstash/spec/outputs/loki_spec.rb

```diff
@@ -0,0 +1,103 @@
+# encoding: utf-8
+require "logstash/devutils/rspec/spec_helper"
+require "logstash/outputs/loki"
+require "logstash/codecs/plain"
+require "logstash/event"
+require "net/http"
+
```

Comment:
perhaps add test for the retry and buffering mechanisms.

Reply:
Sure

Reply:
This is looking good.